packages/flutter/lib/src/material/switch_list_tile.dart

```diff
@@ -187,19 +226,37 @@ class SwitchListTile extends StatelessWidget {
   /// Normally, this property is left to its default value, false.
   final bool selected;
 
+  /// If adaptive, creates the switch with [Switch.adaptive].
+  final _SwitchListTileType _switchListTileType;
+
   @override
   Widget build(BuildContext context) {
-    final Widget control = Switch(
-      value: value,
-      onChanged: onChanged,
-      activeColor: activeColor,
-      activeThumbImage: activeThumbImage,
-      inactiveThumbImage: inactiveThumbImage,
-      materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
-      activeTrackColor: activeTrackColor,
-      inactiveTrackColor: inactiveTrackColor,
-      inactiveThumbColor: inactiveThumbColor,
-    );
+    Widget control;
+    if (_switchListTileType == _SwitchListTileType.adaptive) {
```

Comment:
In Flutter we use switch to dispatch on enums: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values